apps/studio/src/server/modules/folder/folder.router.ts

```diff
@@ -130,33 +151,71 @@ export const folderRouter = router({
           userId: ctx.user.id,
         })
 
-        const result = await db
-          .updateTable("Resource")
-          .where("Resource.id", "=", resourceId)
-          .where("Resource.siteId", "=", Number(siteId))
-          .where("Resource.type", "in", [
-            ResourceType.Folder,
-            ResourceType.Collection,
-          ])
-          .set({
-            permalink,
-            title,
-          })
-          .returning(defaultFolderSelect)
-          .executeTakeFirst()
-          .catch((err) => {
-            if (get(err, "code") === PG_ERROR_CODES.uniqueViolation) {
-              throw new TRPCError({
-                code: "CONFLICT",
-                message: "A resource with the same permalink already exists",
-              })
-            }
-            throw err
+        const result = await db.transaction().execute(async (tx) => {
+          const user = await tx
+            .selectFrom("User")
+            .where("id", "=", ctx.user.id)
+            .selectAll()
+            .executeTakeFirstOrThrow(() => new TRPCError({ code: "NOT_FOUND" }))
+
+          const oldResource = await db
+            .selectFrom("Resource")
+            .selectAll()
+            .where("Resource.id", "=", resourceId)
+            .where("Resource.siteId", "=", Number(siteId))
+            .where("Resource.type", "in", [
+              ResourceType.Folder,
+              ResourceType.Collection,
+            ])
+            .executeTakeFirst()
+
+          if (!oldResource) {
+            return undefined
+          }
```

Comment:
[issue] won't this just bubble up as `undefined`? i think better to just throw `NOT_FOUND` ba

Reply:
Updated in 2ba86e239e4eabc16804821831254b47a74b7e85.